src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1219,79 +1219,90 @@ def apply_fixes(
         after = []
         fixes_applied: List[LintFix] = []
         todo_buffer = list(self.segments)
-        while True:
-            if len(todo_buffer) == 0:
-                break
-            else:
-                seg = todo_buffer.pop(0)
+        requires_validate = False
+
+        while todo_buffer:
```

Comment:
rather than `while True: ... if todo_buffer...`, better to just have `while todo_buffer:`